src/test/java/brave/http/ITRatpackHttpClient.java

```diff
@@ -13,138 +13,215 @@
-                .byteBufAllocator(UnpooledByteBufAllocator.DEFAULT)
-                .maxContentLength(ServerConfig.DEFAULT_MAX_CONTENT_LENGTH)));
-        }).getValue());
+  private static ExecHarness harness;
+  private static AtomicReference<MutableRegistry> registry = new AtomicReference<>();
+
+  @Override protected CurrentTraceContext.Builder currentTraceContextBuilder() {
+    return RatpackCurrentTraceContext.newBuilder().registrySupplier(registry::get);
+  }
+
+  @BeforeClass
+  public static void beforeAll() {
+    harness = ExecHarness.harness();
+  }
+
+  @AfterClass
+  public static void afterAll() {
+    harness.close();
+  }
+
+  @Override protected HttpClient newClient(int port) {
+    return Exceptions.uncheck(() -> harness.yield(e -> {
+      ClientTracingInterceptor clientTracingInterceptor =
+          new DefaultClientTracingInterceptor(httpTracing, () -> Optional.of(e));
+      return Promise.value(HttpClient.of(s -> s
+          .poolSize(0)
+          .requestIntercept(clientTracingInterceptor::request)
+          .responseIntercept(clientTracingInterceptor::response)
+          .errorIntercept(clientTracingInterceptor::error)
+          .byteBufAllocator(UnpooledByteBufAllocator.DEFAULT)
+          .maxContentLength(ServerConfig.DEFAULT_MAX_CONTENT_LENGTH)));
+    }).getValue());
+  }
+
+  @Override protected void closeClient(HttpClient client) {
+    client.close();
+  }
+
+  @Override protected void get(HttpClient client, String pathIncludingQuery) {
+    try {
+      harness.yield(e -> client.get(URI.create(url(pathIncludingQuery)))).getValueOrThrow();
+    } catch (RuntimeException e) {
+      throw e;
+    } catch (Exception e) {
+      throw new RuntimeException(e);
     }
-
-    @Override protected void closeClient(HttpClient client) throws IOException {
-        client.close();
+  }
+
+  @Override protected void post(HttpClient client, String pathIncludingQuery, String body) {
+    try {
+      harness.yield(e ->
+          client.post(URI.create(url(pathIncludingQuery)), (request ->
+              request.body(b -> b.text(body))
+          ))).getValueOrThrow();
+    } catch (RuntimeException e) {
+      throw e;
+    } catch (Exception e) {
+      throw new RuntimeException(e);
     }
-
-    @Override protected void get(HttpClient client, String pathIncludingQuery) throws Exception {
-        harness.yield(e -> client.get(URI.create(url(pathIncludingQuery)))).getValueOrThrow();
-    }
-
-    @Override protected void post(HttpClient client, String pathIncludingQuery, String body)
-        throws Exception {
-        harness.yield(e ->
-            client.post(URI.create(url(pathIncludingQuery)), (request ->
-                request.body(b -> b.text(body))
-            ))
-        ).getValueOrThrow();
+  }
+
+  @Override
+  protected void get(HttpClient client, String path, BiConsumer<Integer, Throwable> callback) {
+    try {
+      harness.run(e ->
```

Comment:
intention here is to not block on run() to complete.. not sure if that's the case.